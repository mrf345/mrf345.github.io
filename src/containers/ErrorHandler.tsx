import React from 'react'
import { Container, Row, Col } from 'react-bootstrap'


interface ErrorHandlerState {
    errors:Error[]
}


export default class ErrorHandler extends React.Component<any, ErrorHandlerState> {
    constructor(props:any) {
        super(props)

        this.state = {errors: []}
    }

    componentDidCatch(err:Error) {
        this.setState({errors: [...this.state.errors, err]})
    }

    render() {
        return this.state.errors.length
            ? <Container className="text-center">
                <Row>
                    <Col className="mb-3 mt-3">
                        <h1>Oh snap! Something went wrong ðŸ˜­</h1>
                    </Col>
                </Row>
                {this.state.errors.map((err, index) => <Row>
                    <Col className="mt-2">
                        <h3>{index + 1}. {err.name}: {err.message}</h3>
                        <p><i>{err.stack}</i></p>
                    </Col>
                </Row>)}
            </Container>
            : this.props.children
    }
}
