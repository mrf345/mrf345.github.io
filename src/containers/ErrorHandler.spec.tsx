import React from 'react'
import { render } from '@testing-library/react'

import ErrorHandler from './ErrorHandler'


describe("Testing ErrorHandler container", () => {
    const FalsyComponent = (props:any) => {
        const err = new Error('Fails to render!')
        err.stack = ''
        throw err

        return <h1>Should Fail!</h1>
    }

    test('Sanity check ErrorHandler contents', () => {
        const component = render(
            <ErrorHandler>
                <FalsyComponent />
            </ErrorHandler>
        )

        expect(component).toMatchSnapshot()
    })
})
