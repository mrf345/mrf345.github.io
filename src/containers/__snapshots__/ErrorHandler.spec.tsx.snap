// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ErrorHandler container Sanity check ErrorHandler contents 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-center container"
      >
        <div
          class="row"
        >
          <div
            class="mb-3 mt-3 col"
          >
            <h1>
              Oh snap! Something went wrong 
              <span
                aria-label="crying!"
                role="img"
              >
                ðŸ˜­
              </span>
            </h1>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="mt-2 col"
          >
            <h3>
              1
              . 
              Error
              : 
              Fails to render!
            </h3>
            <p>
              <i />
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-center container"
    >
      <div
        class="row"
      >
        <div
          class="mb-3 mt-3 col"
        >
          <h1>
            Oh snap! Something went wrong 
            <span
              aria-label="crying!"
              role="img"
            >
              ðŸ˜­
            </span>
          </h1>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="mt-2 col"
        >
          <h3>
            1
            . 
            Error
            : 
            Fails to render!
          </h3>
          <p>
            <i />
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
